app-cpp/app355.cpp

```diff
@@ -182,71 +418,115 @@ std::list<std::list<OrbitalData>> CreateTrains(std::list<OrbitalData> orbitalLis
         prevInclination = data.GetTLE().GetInclination();
 
         newTrain.push_back(data);
-    }
+    });
     if (!newTrain.empty() && newTrain.size() > 2)
     {
-        trainList.push_back(newTrain);
+        trainVector.push_back(newTrain);
     }
 
-    return trainList;
+    // Some trains will be in close proximity, therefore we must merge them
+    // Merge trains whose satellites' mean motions are within 0.001 degrees of each other
+    for (std::size_t i = 0; i < trainVector.size(); ++i)
+    {
+        for (std::size_t j = i + 1; j < trainVector.size(); ++j)
+        {
+            double deltaMotion = abs(trainVector[i][0].GetTLE().GetMeanMotion() - trainVector[j][0].GetTLE().GetMeanMotion());
+            if (deltaMotion < 0.001)
+            {
+                trainVector[i].insert(trainVector[i].end(), trainVector[j].begin(), trainVector[j].end());
+                trainVector.erase(trainVector.begin() + j);
+                --j;
+            }
+        }
+    }
+    
+    return trainVector;
 }
 
-// Section 4: Print Trains
-void PrintTrains(std::list<std::list<OrbitalData>> trainList)
+void SatOrbit::OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainVector)
 {
     int trainCount = 0;
     // Print the contents of the train list
-    for (auto& train : trainList)
+
+    std::for_each(trainVector.begin(), trainVector.end(), [&](auto& train)
     {
         std::cout << "   TRAIN #" << trainCount << std::endl;
         std::cout << "   COUNT: " << train.size() << std::endl;
-    
-        for (auto& data : train)
+
+        std::for_each(train.begin(), train.end(), [](auto& data)
         {
             // Print the name, mean motion, latitude, longitude, and altitude
             std::cout << data.GetTLE().GetName() << ": " << data.GetTLE().GetMeanMotion() << std::endl;
             std::cout << "Lat: " << data.GetLatitude() << std::endl;
             std::cout << "Lon: " << data.GetLongitude() << std::endl;
             std::cout << "Alt: " << data.GetAltitude() << std::endl << std::endl;
-        }
+        });
         std::cout << std::endl << std::endl;
         trainCount++;
-    }
+    });
+}
+
+} // namespace anonymous
+
+class Timer
```

Comment:
`class Timer` is a class local to this `.cpp` and will never be used elsewhere.
Therefore: "it should be placed inside anonymous namespace".
Otherwise, the compiler will needlessly add all the `class Timer` symbol information (that no one else will ever use) into the result `.obj` file... And then that will slow down the linker (with unused/unneeded symbols) when it goes to combine all the `.obj`s into the final `.exe`.